### ISSUE TITLE
`prefer-top-level-await` false positive in case of expressions in `Promise` combinators

### ISSUE THREAD

**Issue by zloirock** [sindresorhus/eslint-plugin-unicorn#2380]
```js
async function getStat(pkg) {
  // ...
}

const [foo, bar] = await Promise.all([
  getStat('foo'),
  CONDITION && getStat('bar'), // that trigger `prefer-top-level-await`
]);
```

**Comment by jhnns** [comment]
I think that's a duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2177 ? :)

**Comment by zloirock** [comment]
It seems so.
